File: week1-cpp/Basic_Image_Operations/Create_new_images.cpp
// Include libraries
#include <iostream>
#include "dataPath.hpp"
#include <opencv2/opencv.hpp>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>

using namespace std;
using namespace cv;

int main(void){
	// Read image
	Mat image = imread(DATA_PATH+"images/boy.jpg");
	//imshow("Input Image",image);
	//waitKey(0);
	
	// Create a new image by copying the already present image 
	// using the clone operation
	Mat imageCopy = image.clone();

	Mat emptyMatrix = Mat(100,200,CV_8UC3, Scalar(0,0,0));
	imwrite("../results/emptyMatrix.png",emptyMatrix);
	
	//imshow("Empty Matrix",emptyMatrix);
	//waitKey(0);
	
	emptyMatrix.setTo(Scalar(255,255,255));
	imwrite("../results/emptyMatrixWhite.png",emptyMatrix);
	//imshow("Empty Matrix White",emptyMatrix);

	Mat emptyOriginal = Mat(emptyMatrix.size(), 
			emptyMatrix.type(), Scalar(100,100,100));
	imwrite("../results/emptyMatrix100.png",emptyOriginal);
	//imshow("Empty Original",emptyOriginal);
	
	return 0;
}

File: week1-python/Basic_Image_Operations/Create_New_Images.py
# Import libraries
import cv2
import numpy as np
from dataPath import DATA_PATH

# Read image
image = cv2.imread(DATA_PATH+"images/boy.jpg")

# Let's see what image we are dealing with
#cv2.imshow("Input image",image)

# Create a new image by copying the already present image using the copy operation
imageCopy = image.copy()

# Create an empty matrix
emptyMatrix = np.zeros((100,200,3),dtype='uint8')
#cv2.imshow("Empty matrix",emptyMatrix)
#cv2.waitKey(0)
cv2.imwrite("../results/emptyMatrix.png",emptyMatrix)

emptyMatrix = 255*np.ones((100,200,3),dtype='uint8')
#cv2.imshow("Empty matrix",emptyMatrix)
cv2.imwrite("../results/emptyMatrix1.png",emptyMatrix)

emptyOriginal = 100*np.ones_like(image)
#cv2.imshow("Empty Original",emptyOriginal)
#cv2.waitKey(0)
#cv2.destroyAllWindows()
cv2.imwrite("../results/emptyOriginal.png",emptyOriginal)
